axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveCounterPartServiceImpl.java

```diff
@@ -80,53 +85,35 @@ protected BigDecimal getCounterpartAmount(Move move) {
-          .getJournalType()
-          .getTechnicalTypeSelect()
-          .equals(JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE)) {
-        accountingAccount = accountingSituation.getSupplierAccount();
-      } else if (move.getJournal()
-          .getJournalType()
-          .getTechnicalTypeSelect()
-          .equals(JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE)) {
-        accountingAccount = accountingSituation.getCustomerAccount();
+    Company company = move.getCompany();
+    int technicalTypeSelect = move.getJournal().getJournalType().getTechnicalTypeSelect();
+    if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE
+        || technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE) {
+      if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE) {
+        accountingAccount =
+            accountingSituationService.getSupplierAccount(move.getPartner(), move.getCompany());
+      } else if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE) {
+        accountingAccount =
+            accountingSituationService.getCustomerAccount(move.getPartner(), move.getCompany());
       }
 
-    } else if (move.getJournal()
-        .getJournalType()
-        .getTechnicalTypeSelect()
-        .equals(JournalTypeRepository.TECHNICAL_TYPE_SELECT_TREASURY)) {
-      if (move.getPaymentMode() != null)
-        for (AccountManagement accountManagement :
-            move.getPaymentMode().getAccountManagementList()) {
-          if (accountManagement.getCompany().equals(move.getCompany())) {
-            accountingAccount = accountManagement.getCashAccount();
-          }
+    } else if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_TREASURY) {
+      if (move.getPaymentMode() != null) {
+        AccountManagement accountManagement =
+            accountManagementService.getAccountManagement(
+                move.getPaymentMode().getAccountManagementList(), company);
+        if (ObjectUtils.notEmpty(accountManagement)) {
+          accountingAccount = accountManagement.getCashAccount();
         }
+      }
     }
     if (accountingAccount == null) {
       AccountConfig accountConfig = accountConfigService.getAccountConfig(move.getCompany());
-      if (move.getJournal()
-          .getJournalType()
-          .getTechnicalTypeSelect()
-          .equals(JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE))
+      if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_EXPENSE) {
         accountingAccount = accountConfigService.getSupplierAccount(accountConfig);
-      else if (move.getJournal()
-          .getJournalType()
-          .getTechnicalTypeSelect()
-          .equals(JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE))
+      } else if (technicalTypeSelect == JournalTypeRepository.TECHNICAL_TYPE_SELECT_SALE) {
         accountingAccount = accountConfigService.getCustomerAccount(accountConfig);
```

Comment:
accountingAccount =
accountingSituationService.getSupplierAccount(move.getPartner(), move.getCompany());

In this method, if account is null, we already get the account from accountConfig. So I think no need this statements.